### PR TITLE
cmd/systray: add account switcher

### PR DESCRIPTION
**Pull request by awly**
Updates #1708

### CODE REVIEW

**Comment by awly** [comment]
@willnorris I believe I figured out the empty submenu bug. As expected, it's a race condition over dbus that can be "fixed" with a good ol' sleep.

I ran `waybar -l trace` and noticed that every time the bug happens, it prints a bunch of:
```
(waybar:153009): LIBDBUSMENU-GTK-WARNING **: 18:07:11.551: Children but no menu, someone's been naughty with their 'children-display' property: 'submenu'
```
which comes from https://bazaar.launchpad.net/~dbusmenu-team/libdbusmenu/trunk.16.04/view/head:/libdbusmenu-gtk/client.c#L965

My understanding is that `libdbusmenu-gtk` uses some global object store thingy from GTK to keep track of various menu objects. When it receives a dbus message about the child menu item and doesn't find the parent in that store, it prints the warning and bails.

On my machine, delaying child menu items by 100ms after the parent is created seems to fix it, at least within 10 consecutive systray restarts.

LMK what you think!

File: cmd/systray/systray.go [hunk 1]
```diff
@@ -64,39 +66,97 @@ func onReady() {
...
+	//    (waybar:153009): LIBDBUSMENU-GTK-WARNING **: 18:07:11.551: Children but no menu, someone's been naughty with their 'children-display' property: 'submenu'
+	time.Sleep(100 * time.Millisecond)
+	// Aggregate all clicks into a shared channel.
+	menu.accountsCh = make(chan ipn.ProfileID)
+	for _, profile := range state.allProfiles {
+		title := fmt.Sprintf("%s (%s)", profile.Name, profile.NetworkProfile.DomainName)
+		// Note: we could use AddSubMenuItemCheckbox instead of this formatting
+		// hack, but checkboxes don't work across all desktops unfortunately.
+		if profile.ID == state.curProfile.ID {
+			title = "* " + title
```
Comment (willnorris) [inline cmd/systray/systray.go:139]: for future improvement, here is what I was doing to display tailnet name only when the profile name was ambiguous, as well as to display profile photos inline: https://github.com/tailscale/tailscale/blob/will-systray/cmd/systray/systray.go#L197

somewhere I had an alternate implementation that did caching of the http responses, but I'll have to find it.

**Review by willnorris — APPROVED** [review]
oh man... great find!  This seems to have fixed it!  This looks totally fine to add for now.  I left a link below to some previous code for including profile pictures as well, but I may have some better code laying around somewhere as well.

In any event, now that you've figured out the submenu issue, this should make iterating on this much easier!

File: cmd/systray/systray.go (hunk 1 above)
Comment (willnorris) [reply cmd/systray/systray.go:139]: heh, actually it looks like that does have the caching code in https://github.com/tailscale/tailscale/blob/will-systray/cmd/systray/systray.go#L263-L280, but is also doing the non-caching version elsewhere.  Some of that code was slightly trashy, so should just be treated as a reference if it's still valuable as such.

**Comment by willnorris** [comment]
oh, I also forgot to mention that the systray icon doesn't seem to go back to the loading animation when switching accounts.  Maybe need to call `setAppIcon(loading)` as soon as the profile switch is initiated?  I would have thought that the state change would trigger it, but maybe that's happening too late?

**Comment by willnorris** [comment]
I'm going to go ahead and merge this as-is so I can hopefully built on top of it over the holidays.

**Pull request merged**